### PR TITLE
Enhance node interface

### PR DESCRIPTION
**Pull request by fyang1024**

### CODE REVIEW

File: pallets/deeper-node/src/lib.rs [hunk 1]
```diff
@@ -745,9 +751,20 @@ pub mod pallet {
...
+            let current_block = <frame_system::Module<T>>::block_number();
+            current_block - block >= T::DayToBlocknum::get().into()
+        }
+
+        fn im_ever_online(account_id: &T::AccountId) -> bool {
+            Self::get_im_online(account_id) != None
+        }
+
+        fn get_days_offline(account_id: &T::AccountId) -> u32 {
+            let block = Self::get_im_online(account_id).unwrap_or(T::BlockNumber::default());
```
Comment (lzw65) [inline pallets/deeper-node/src/lib.rs:764]: Should we add a checkout condition : 
`if Self::im_ever_online（account_id）== false {
 return 0u32}`

File: pallets/deeper-node/src/tests.rs [hunk 2]
```diff
@@ -183,7 +183,29 @@ fn im_offline() {
...
+#[test]
+fn get_days_offline() {
+    new_test_ext().execute_with(|| {
+        assert_ok!(DeeperNode::im_online(Origin::signed(1)));
+        run_to_block(24 * 3600 * 1000 / 5000 - 1);
+        assert_eq!(DeeperNode::get_days_offline(&1), 0);
+        run_to_block(24 * 3600 * 1000 / 5000);
+        assert_eq!(DeeperNode::get_days_offline(&1), 1);
+        run_to_block(3 * 24 * 3600 * 1000 / 5000);
+        assert_eq!(DeeperNode::get_days_offline(&1), 3);
```
Comment (lzw65) [inline pallets/deeper-node/src/tests.rs:209]: Add a case:
`assert_eq!(DeeperNode::get_days_offline(&2), 0);`

File: pallets/deeper-node/src/lib.rs (hunk 1 above)
Comment (fyang1024) [reply pallets/deeper-node/src/lib.rs:764]: Yes, I thought about this. It really depends on how we gonna interpret this.
Current implementation means if a device is never online, we deem it as offline. There should be consistency between the `im_offline` and `get_days_offline` logic. If `im_offline` returns `true`, `get_days_offline` >= `1`, and vice versa. If `im_offline` returns false, `get_days_offline` = `0`, and vice versa.

My thinking is, if `im_ever_online` = `false`, it makes logical sense for `im_offline` = `true`

Please let me know if there is any reason that invalidates the logic

File: pallets/deeper-node/src/tests.rs (hunk 2 above)
Comment (fyang1024) [reply pallets/deeper-node/src/tests.rs:209]: Yes, we need to add one more assertions here, but it is related to the pervious comments. It should be
`assert_eq!(DeeperNode::get_days_offline(&2), 3);` with current interpretation.

Comment (fyang1024) [reply pallets/deeper-node/src/tests.rs:209]: As discussed, I've added one more assertion to test the case when a device is never online

**Review by lzw65 — APPROVED** [review]

**Pull request merged**